Kernel/Bus/PCI/Controller/HostController.cpp

```diff
@@ -128,10 +148,120 @@ UNMAP_AFTER_INIT void HostController::enumerate_attached_devices(Function<void(E
                 break;
             if (m_enumerated_buses.get(m_domain.start_bus() + bus_as_function_number))
                 continue;
-            enumerate_bus(callback, m_domain.start_bus() + bus_as_function_number, false);
+            enumerate_bus(callback, post_bridge_callback, m_domain.start_bus() + bus_as_function_number, false);
             m_enumerated_buses.set(m_domain.start_bus() + bus_as_function_number, true);
         }
     }
 }
 
+void HostController::configure_attached_devices(FlatPtr& mmio_32bit_base, FlatPtr mmio_32bit_end, FlatPtr& mmio_64bit_base, FlatPtr mmio_64bit_end)
+{
+    // First, Assign PCI-to-PCI bridge bus numbering
+    u8 bus_id = 0;
+    enumerate_attached_devices([this, &bus_id](EnumerableDeviceIdentifier const& device_identifier) {
```

Comment:
The indentations of the lambdas in the two calls to enumerate_attached_devices are a bit confusing, not sure if you can get clang-format to format them a bit nicer.

Reply:
Yeah, this seems better now.